logmetric.go

```diff
@@ -4,24 +4,24 @@ import (
 	"sync"
 	"time"
 
+	"github.com/rs/zerolog"
 	"go.opentelemetry.io/otel/attribute"
-	"go.uber.org/zap"
 )
 
 type LogMetric struct {
-	mu         sync.RWMutex // Allows multiple reads, blocks only on writes
-	fields     map[string]zap.Field
+	mu         sync.RWMutex
+	fields     map[string]any
 	attributes map[string]attribute.KeyValue
 }
 
 // NewLogMetric initializes a LogMetric instance
-func NewLogMetric(fields []zap.Field, attributes []attribute.KeyValue) *LogMetric {
+func NewLogMetric(initialFields map[string]any, attributes []attribute.KeyValue) *LogMetric {
```

Comment:
do we still need logMetric?
logMetric if i remember isn't too efficient and log.Info().Fields(...) can easily be overwritten.

Reply:
maybe we'll do that later and keep an eye on this.